activerecord/lib/active_record/statement_cache.rb

```diff
@@ -106,6 +106,8 @@ def execute(params, connection, &block)
       sql = query_builder.sql_for bind_values, connection
 
       klass.find_by_sql(sql, bind_values, preparable: true, &block)
+    rescue ::RangeError
+      []
```

Comment:
This rescue makes me nervous, just because it *sounds* far-reaching.

I agree it's fine to put it here in practice, because of the very limited set of statements that are eligible for this cache, as you noted in the commit message. I think that's worth a comment here too, though: both to hopefully draw attention if the cache ever gets used for other queries, and just to reassure ~future readers~ me next time I read through this code 😅